tests/unit/h/cli/commands/init_test.py

```diff
@@ -1,92 +1,124 @@
-        db_engine.execute("CREATE TABLE alembic_version (version_num VARCHAR(32));")
+        caplog.set_level(logging.INFO)
 
-        @request.addfinalizer
-        def _cleanup():
-            db_engine.execute("DROP TABLE alembic_version;")
+        result = cli.invoke(init, obj=ctx)
 
-        result = cli.invoke(init_cli.init, obj=cliconfig)
-
-        assert not db.init.called
+        db.init.assert_not_called()
+        alembic.command.stamp.assert_not_called()
+        assert (
+            "h.cli.commands.init",
+            logging.INFO,
+            "detected alembic_version table, skipping db initialization",
+        ) in caplog.record_tuples
         assert not result.exit_code
 
-    def test_stamps_alembic_version(
-        self,
-        alembic_config,
-        alembic_stamp,
-        cli,
-        cliconfig,
-        db,
-        db_engine,
-        pyramid_settings,
+    @pytest.mark.parametrize(
+        "settings,check_icu_plugin",
+        [
+            ({"es.check_icu_plugin": True}, True),
+            ({"es.check_icu_plugin": False}, False),
+            ({}, True),
+        ],
+    )
+    def test_it_initializes_elasticsearch(
+        self, caplog, cli, ctx, search, pyramid_request, settings, check_icu_plugin
     ):
-        db.make_engine.return_value = db_engine
-        Config = alembic_config.Config
-        pyramid_settings["h.authority"] = "foobar.org"
-
-        result = cli.invoke(init_cli.init, obj=cliconfig)
-
-        Config.assert_called_once_with("conf/alembic.ini")
-        alembic_stamp.assert_called_once_with(Config.return_value, "head")
+        caplog.set_level(logging.INFO)
+        pyramid_request.registry.settings.update(settings)
+
+        result = cli.invoke(init, obj=ctx)
+
+        search.get_client.assert_called_once_with(pyramid_request.registry.settings)
+        assert (
+            "h.cli.commands.init",
+            logging.INFO,
+            "initializing ES6 search index",
+        ) in caplog.record_tuples
+        search.init.assert_called_once_with(
+            search.get_client.return_value, check_icu_plugin=check_icu_plugin
+        )
         assert not result.exit_code
 
-    def test_initialises_search(self, cli, cliconfig, search, pyramid_settings):
-        pyramid_settings["es.check_icu_plugin"] = False
-        es_client = search.get_client.return_value
-        result = cli.invoke(init_cli.init, obj=cliconfig)
 
-        search.get_client.assert_called_once_with(pyramid_settings)
-        search.init.assert_any_call(es_client, pyramid_settings["es.check_icu_plugin"])
-        assert not result.exit_code
+@pytest.fixture(autouse=True)
+def os(patch):
+    os = patch("h.cli.commands.init.os")
+    os.environ = {}
+    return os
 
 
-@pytest.fixture
-def cliconfig(pyramid_request):
-    return {"bootstrap": mock.Mock(return_value=pyramid_request)}
+@pytest.fixture(autouse=True)
+def alembic(patch):
+    return patch("h.cli.commands.init.alembic")
```

Comment:
Patches should always be within the namespace of the module-under-test itself, so we're patching the name `alembic` in the `h.cli.commands.init` module, not patching sub-names within the `h.cli.commands.init.alembic` module